modules/acm_sku/src/CategoryManager.php

```diff
@@ -357,8 +357,18 @@ private function syncCategory(array $categories, $parent = NULL, $storeId = '')
         // Break child relationships.
         $children = $this->termStorage->loadChildren($term->id(), $this->vocabulary->id());
         if (count($children)) {
-          $child_ids = array_map(function ($child) {
-            return ($child->id());
+          $child_ids = array_map(function ($child) use ($category) {
+            // If term having commerce id, means its sync from magento and
+            // thus we process. Term not having commerce id means its created
+            // only on Drupal and thus we skip processing.
+            if ($commerce_id = $child->get('field_commerce_id')->first()) {
+              // We check if the child exists in the response get from magento.
+              foreach ($category['children'] as $sync_cat_child) {
```

Comment:
should this be `$child['children']`?

Reply:
No, this should be $category[children] as we checking if the term we get from magento has children same as as we load the term from Drupal and its children